test/mod/test_buildroot.py

```diff
@@ -135,5 +143,41 @@ def test_selinuxfs_ro(self):
                    "/sys/fs/selinux",
                    "ro"]
 
-            r = root.run(cmd, monitor, readonly_binds=ro_binds)
+            r = root.run(cmd, monitor, timeout, readonly_binds=ro_binds)
             self.assertEqual(r.returncode, 0)
+
+    def test_timeout(self):
+        runner = detect_host_runner()
+        libdir = os.path.abspath(os.curdir)
+        var = pathlib.Path(self.tmp.name, "var")
+        var.mkdir()
+
+        with BuildRoot("/", runner, libdir, var) as root:
+
+            READ_ONLY = select.POLLIN | select.POLLPRI | select.POLLHUP | select.POLLERR
+            poller = select.poll()
+
+            proc = subprocess.Popen(['python3', libdir + '/stages/org.osbuild.test.timeout'], stdout = subprocess.PIPE)
+            start = time.monotonic()
+            timeout = 4
+            poller.register(proc.stdout.fileno(), READ_ONLY)
+            r = root.read_with_timeout(proc, poller, start, timeout)
+            self.assertEqual(r, b'Test stage output.\n')
+
+            proc = subprocess.Popen(['python3', libdir + '/stages/org.osbuild.test.timeout'], stdout = subprocess.PIPE)
+            start = time.monotonic()
+            timeout = 0
+            poller.register(proc.stdout.fileno(), READ_ONLY)
+            try:
+                r = root.read_with_timeout(proc, poller, start, timeout)
+                self.assertEqual(r, b'Test stage output.\n')
+            except TimeoutError:
+                self.assertEqual(0, 0)
```

Comment:
This is triggered if the timeout is hit, which it should be, but why not have stage hang forever with a 1 s timeout and assert that the timeout is indeed triggered?